moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -150,33 +156,37 @@ PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParamet
   if (current_goal_constraints_.empty())
   {
     RCLCPP_ERROR(LOGGER, "No goal constraints set for planning request");
-    last_plan_solution_->error_code = moveit::core::MoveItErrorCode::INVALID_GOAL_CONSTRAINTS;
-    return *last_plan_solution_;
+    last_plan_solution_.error_code_ = moveit::core::MoveItErrorCode::INVALID_GOAL_CONSTRAINTS;
+    return last_plan_solution_;
   }
   req.goal_constraints = current_goal_constraints_;
 
   // Set path constraints
   req.path_constraints = current_path_constraints_;
+  // Set trajectory constraints
+  req.trajectory_constraints = current_trajectory_constraints_;
 
   // Run planning attempt
   ::planning_interface::MotionPlanResponse res;
   if (planning_pipeline_names_.find(parameters.planning_pipeline) == planning_pipeline_names_.end())
   {
     RCLCPP_ERROR(LOGGER, "No planning pipeline available for name '%s'", parameters.planning_pipeline.c_str());
-    last_plan_solution_->error_code = moveit::core::MoveItErrorCode::FAILURE;
-    return *last_plan_solution_;
+    last_plan_solution_.error_code_ = moveit::core::MoveItErrorCode::FAILURE;
+    return last_plan_solution_;
   }
   const planning_pipeline::PlanningPipelinePtr pipeline =
       moveit_cpp_->getPlanningPipelines().at(parameters.planning_pipeline);
   pipeline->generatePlan(planning_scene, req, res);
-  last_plan_solution_->error_code = res.error_code_.val;
+  last_plan_solution_.error_code_ = res.error_code_.val;
```

Comment:
Why is there a .val here? I thought these would be the same type and we could just copy the struct completely.